napi/src/options.rs

```diff
@@ -0,0 +1,274 @@
+    ///
+    /// Default `["node_modules"]`
+    pub modules: Option<Vec<String>>,
+
+    /// Resolve to a context instead of a file.
+    ///
+    /// Default `false`
+    pub resolve_to_context: Option<bool>,
+
+    /// Prefer to resolve module requests as relative requests instead of using modules from node_modules directories.
+    ///
+    /// Default `false`
+    pub prefer_relative: Option<bool>,
+
+    /// Prefer to resolve server-relative urls as absolute paths before falling back to resolve in ResolveOptions::roots.
+    ///
+    /// Default `false`
+    pub prefer_absolute: Option<bool>,
+
+    /// A list of resolve restrictions to restrict the paths that a request can be resolved on.
+    ///
+    /// Default `[]`
+    pub restrictions: Option<Vec<Restriction>>,
+
+    /// A list of directories where requests of server-relative URLs (starting with '/') are resolved.
+    /// On non-Windows systems these requests are resolved as an absolute path first.
+    ///
+    /// Default `[]`
+    pub roots: Option<Vec<String>>,
+
+    /// Whether to resolve symlinks to their symlinked location.
+    /// When enabled, symlinked resources are resolved to their real path, not their symlinked location.
+    /// Note that this may cause module resolution to fail when using tools that symlink packages (like npm link).
+    ///
+    /// Default `true`
+    pub symlinks: Option<bool>,
+
+    /// Whether to parse [module.builtinModules](https://nodejs.org/api/module.html#modulebuiltinmodules) or not.
+    /// For example, "zlib" will throw [crate::ResolveError::Builtin] when set to true.
+    ///
+    /// Default `false`
+    pub builtin_modules: Option<bool>,
+}
+
+#[napi]
+#[derive(Debug, PartialEq, Eq)]
+pub enum EnforceExtension {
+    Auto,
+    Enabled,
+    Disabled,
+}
+
+impl EnforceExtension {
+    pub fn is_auto(&self) -> bool {
+        *self == Self::Auto
+    }
+
+    pub fn is_enabled(&self) -> bool {
+        *self == Self::Enabled
+    }
+
+    pub fn is_disabled(&self) -> bool {
+        *self == Self::Disabled
+    }
+}
+
+/// Alias Value for [ResolveOptions::alias] and [ResolveOptions::fallback].
+/// Use struct because napi don't support structured union now
+#[napi(object)]
+#[derive(Debug, Clone)]
+pub struct Restriction {
+    pub path: Option<String>,
+    pub regex: Option<String>,
+}
+
+/// Tsconfig Options
+///
+/// Derived from [tsconfig-paths-webpack-plugin](https://github.com/dividab/tsconfig-paths-webpack-plugin#options)
+
+#[napi(object)]
+#[derive(Debug, Clone)]
+pub struct TsconfigOptions {
+    /// Allows you to specify where to find the TypeScript configuration file.
+    /// You may provide
+    /// * a relative path to the configuration file. It will be resolved relative to cwd.
+    /// * an absolute path to the configuration file.
+    pub config_file: String,
+    /// Support for Typescript Project References.
+    /// For the type TsconfigReferences
```

Comment:
What does `For the type TsconfigReferences` mean?

Reply:
Use `Either<String, Vec<String>>` instead of `TsconfigReference` because napi don't support structured enum (a variant that doesn't hold any data) .